File: go_algorithms/algorithms/ColumnByColumnSequential.go
package algorithms

import (
	"fmt"
	"time"
)

func ColumnByColumnSequential(A, B [][]int) ([][]int, int64) {

	fmt.Println("Método: ColumnByColumnSequential")

	startTime := time.Now().UnixNano()

	rowsA, colsA, colsB := len(A), len(A[0]), len(B[0])

	result := make([][]int, rowsA)
	for i := range result {
		result[i] = make([]int, colsB)
	}

	for j := 0; j < colsB; j++ {
		for i := 0; i < rowsA; i++ {
			for k := 0; k < colsA; k++ {
				result[i][j] += A[i][k] * B[k][j]
			}
		}
	}

	endTime := time.Now().UnixNano()
	executionTime := endTime - startTime
	return result, executionTime
}

File: python_algorithms/algorithms/ColumnByColumnSequential.py
import numpy as np
import time

def Multiply(A, B):
    print("Método: ColumnByColumnSequential")

    start_time = time.time_ns()

    rowsA, colsA, colsB = len(A), len(A[0]), len(B[0])

    result = np.zeros((rowsA, colsB), dtype=np.int64)

    for j in range(colsB):
        for i in range(rowsA):
            for k in range(colsA):
                result[i][j] += A[i][k] * B[k][j]

    end_time = time.time_ns()
    execution_time = end_time - start_time
    return result, execution_time
